lib/src/Base/Optim/MultiStart.cxx

```diff
@@ -0,0 +1,212 @@
+MultiStart::MultiStart(const OptimizationSolver & solver)
+  : OptimizationSolverImplementation(solver.getProblem())
+  , solver_(solver)
+  , startingPointNumber_(ResourceMap::GetAsUnsignedInteger("MultiStart-DefaultStartingPointNumber"))
+{
+  OptimizationProblem problem(getProblem());
+  if (!problem.hasBounds())
+    throw InvalidArgumentException(HERE) << "Cannot use default multi-start without bounds when the experiment is not defined";
+
+  // sample uniformly over the bounds
+  const UnsignedInteger dimension = problem.getDimension();
+  ComposedDistribution::DistributionCollection coll(dimension);
+  for (UnsignedInteger j = 0; j < dimension; ++ j)
+  {
+    coll[j] = Uniform(problem.getBounds().getLowerBound()[j], problem.getBounds().getUpperBound()[j]);
+  }
+  Distribution initialDistribution = ComposedDistribution(coll);
+  experiment_ = MonteCarloExperiment(initialDistribution, 1);
+}
+
+
+/* Constructor with parameters */
+MultiStart::MultiStart(const OptimizationSolver & solver,
+                       const Experiment & experiment)
+  : OptimizationSolverImplementation(solver.getProblem())
+  , solver_(solver)
+  , experiment_(experiment)
+  , startingPointNumber_(ResourceMap::GetAsUnsignedInteger("MultiStart-DefaultStartingPointNumber"))
+{
+}
+
+
+/* Check whether this problem can be solved by this solver.  Must be overloaded by the actual optimisation algorithm */
+void MultiStart::checkProblem(const OptimizationProblem & problem) const
+{
+  // Nothing to do
+}
+
+
+void MultiStart::run()
+{
+  // each starting point is the best of a new exploratory design
+  startingPoints_ = NumericalSample(0, getProblem().getDimension());
+  for (UnsignedInteger i = 0; i < startingPointNumber_; ++ i)
+  {
+    NumericalSample explorationPoints(experiment_.generate());
+    NumericalSample explorationValues(getProblem().getObjective()(explorationPoints));
+    const UnsignedInteger explorationSize = explorationPoints.getSize();
+    NumericalPoint bestExplorationPoint = explorationPoints[0];
+    NumericalScalar bestExplorationValue = explorationValues[0][0];
+    for (UnsignedInteger j = 1; j < explorationSize; ++ j)
+    {
+      if (!getProblem().hasBounds() || (getProblem().hasBounds() && getProblem().getBounds().contains(explorationPoints[j])))
+        if ((getProblem().isMinimization() && (explorationValues[j][0] < bestExplorationValue))
+            || (!getProblem().isMinimization() && (explorationValues[j][0] > bestExplorationValue)))
+        {
+          bestExplorationPoint = explorationPoints[j];
+          bestExplorationValue = explorationValues[j][0];
+        }
+    }
+    startingPoints_.add(bestExplorationPoint);
+  }
+
+  // run the solver with each starting point
+  OptimizationSolver solver(solver_);
+  resultCollection_.clear();
+  NumericalScalar bestValue = getProblem().isMinimization() ? SpecFunc::MaxNumericalScalar : -SpecFunc::MaxNumericalScalar;
+  for (UnsignedInteger i = 0; i < startingPointNumber_; ++ i)
```

Comment:
good idea to initialize the optimal value without any call to the objective function. It should have been used in the previous block of code if it had not to be removed.